gui/src/deadliner.rs

```diff
@@ -443,15 +445,20 @@ fn background_edit(ui: &mut egui::Ui, bg: &mut Background) {
                         let file_ext = file_name.split(".").collect::<Vec<&str>>().pop().unwrap();
                         let supported_file_ext = ["png", "gif", "jpg", "jpeg"];
 
+                        let mut data = ui.data();
+                        let is_valid = data.get_temp_mut_or(ui.id(), IsValid(true));
+                        
                         if supported_file_ext.contains(&file_ext) {
                             *location = new_location;
-                            is_valid = true;
+
+                            *is_valid = IsValid(true);
                         } else {
-                            is_valid = false;
+                            *is_valid = IsValid(false);
```

Comment:
Love it! I was questioning myself for so long if something like this was possible cause I agree with you that it's so messy to store errors, and loading states at the Application level.

Reply:
Rust generated docs are so bad in terms of their searching capabilities. If you also agree that it's so hard to find stuff in rust docs, maybe you can pitch this up:

https://github.com/rust-lang/rust/issues/95213